core/src/commonMain/kotlin/komem/litmus/Utils.kt

```diff
@@ -24,3 +24,35 @@ fun List<List<String>>.tableFormat(hasHeader: Boolean = false): String {
 }
 
 expect fun cpuCount(): Int
+
+/**
+ * A wrapper over Array<Any> that is also a list.
+ */
+@Suppress("UNCHECKED_CAST")
+class CustomList<T>(override val size: Int, initializer: (Int) -> T) : List<T> {
```

Comment:
It does not cast the type. The main limitation for using arrays is that we need them to store states of type `S`, which is generic and not supported by arrays. Hence the need to store `Any?` and cast.

Reply:
> The main limitation for using arrays is that we need them to store states of type S which is generic and not supported by arrays

`Array<T>` is a generic, the only problem is that you need to know the type `T` statically at a site, when you allocate the array. This can be potentially mitigated with the help of `inline reified` functions: 

https://stackoverflow.com/questions/46869353/kotlin-generics-arrayt-results-in-cannot-use-t-as-a-reified-type-parameter-u

Otherwise, I think it is possible to just pass around `Array<Any>`, since arrays are covariant.

Reply:
Agreed to keep as `Array<Any>` with unsafe cast to `Array<S>`